snuba/datasets/table_storage.py

```diff
@@ -57,13 +60,16 @@ def get_physical_topic_name(self, slice_id: Optional[int] = None) -> str:
 
         return physical_topic
 
-    @property
+    @cached_property
     def partitions_number(self) -> int:
-        return settings.TOPIC_PARTITION_COUNTS.get(self.__topic.value, 1)
-
-    @property
-    def replication_factor(self) -> int:
-        return 1
+        config = get_default_kafka_configuration(self.__topic, None)
```

Comment:
This connection already happens in `bootstrap.py`. Is there a way we could leverage that code, to avoid connecting twice on startup?

Also, there is a utility function that does this: https://github.com/getsentry/snuba/blob/master/snuba/consumers/utils.py#L17

Could we use that instead?

Reply:
Ugggh I just noticed a pretty bad bug with that other function. It does not properly resolve the topic name if it is overridden to something else and assumes the default. So I guess we can't safely use it right now, but also... this probably needs to be fixed?

Reply:
Sounds like it should be fixed. Did you want to add it to this PR?